sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala

```diff
@@ -142,7 +142,7 @@ class FileStreamSink(
   }
 
   override def addBatch(batchId: Long, data: DataFrame): Unit = {
-    if (batchId <= fileLog.getLatest().map(_._1).getOrElse(-1L)) {
```

Comment:
Nice catch!

https://github.com/apache/spark/blob/7ad6ba36f28b7a5ca548950dec6afcd61e5d68b9/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala#L196

https://github.com/apache/spark/blob/7ad6ba36f28b7a5ca548950dec6afcd61e5d68b9/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSourceLog.scala#L99

Can these two places also be optimized in this way?

Reply:
Yeah I think so. Nice finding. Thanks!